src/components/shared/SaveBookmark/BookmarkForm.js

```diff
@@ -0,0 +1,209 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { withNamespaces } from 'react-i18next';
+import {
+  Button,
+  Checkbox, Container,
+  Header,
+  Icon,
+  Input,
+  List, Modal,
+  ModalActions,
+  ModalContent,
+  Segment
+} from 'semantic-ui-react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { actions, selectors } from '../../../redux/modules/my';
+import { MY_NAMESPACE_BOOKMARKS, MY_NAMESPACE_FOLDERS } from '../../../helpers/consts';
+import { frownSplashNotFound } from '../WipErr/WipErr';
+import { getMyItemKey } from '../../../helpers/my';
+import NeedToLogin from '../../Sections/Personal/NeedToLogin';
+
+const BookmarkForm = ({ t, onClose, source, bookmarkId, data }) => {
+  const [name, setName] = useState();
+  const [selected, setSelected] = useState(null);
+  const [editFolder, setEditFolder] = useState(false);
+  const [query, setQuery] = useState();
+  const [isEdit, setIsEdit] = useState();
+
+  const { key } = getMyItemKey(MY_NAMESPACE_BOOKMARKS, { id: bookmarkId });
+  const bookmark = useSelector(state => selectors.getItemByKey(state.my, MY_NAMESPACE_BOOKMARKS, key));
+  const items = useSelector(state => selectors.getList(state.my, MY_NAMESPACE_FOLDERS)).filter(x => !query || x.name.toLowerCase().includes(query));
+  const saved = items.filter(f => bookmark?.folder_ids?.includes(f.id)).map(f => f.id);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (items.length === 0)
+      dispatch(actions.fetch(MY_NAMESPACE_FOLDERS, { 'order_by': 'id DESC' }));
+    // eslint-disable-next-line  react-hooks/exhaustive-deps
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (selected === null) {
+      setSelected([...saved]);
+    }
+    // eslint-disable-next-line  react-hooks/exhaustive-deps
```

Comment:
Why disable lint is okay?

Reply:
//update selected only when saved was changed (that mean only on server response)
    //and not on changes of selected

add comment

Reply:
So why not add all the parameters and create the correct condition (if) on line 43?